README.md

```diff
@@ -45,58 +45,140 @@ This project can be utilized one of two ways, via docker or via a Makefile inclu
 a binary from GitHub releases, which you can utilize pretty easily:
 
-```
-## Install gomplate
-gomplate/% GOMPLATE_VERSION ?= latest
-gomplate/install:
-  @ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=hairyhenderson REPO=$(@D) VERSION=$(GOMPLATE_VERSION) QUERY='.name | endswith("$(OS)-$(ARCH)")'
-```
-
-The target `install/gh-release/%` as well as the vars `$(OS)` and `$(ARCH)` are
-provided by the tardigrade-ci/Makefile, and do not need to be redefined in your
-local Makefile.
+    ```
+    ## Install gomplate
+    gomplate/% GOMPLATE_VERSION ?= latest
+    gomplate/install:
+      @ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=hairyhenderson REPO=$(@D) VERSION=$(GOMPLATE_VERSION) QUERY='.name | endswith("$(OS)-$(ARCH)")'
+    ```
+
+    The target `install/gh-release/%` as well as the vars `$(OS)` and `$(ARCH)` are
+    provided by the tardigrade-ci/Makefile, and do not need to be redefined in your
+    local Makefile.
+
+## Makefile prerequisites
+
+Using the Makefile approach involves calling `make` directly. This means make,
+as well as a few other prerequisites must be installed on the system for it to
+work.
+
+* `make` v4.2 or later
+* `jq` v1.5 or later
+* `git` v2.20 or later
+* `bash` v4 or later
+* `sed` v4 or later
+* `awk` v4 or later
+* `grep` v3 or later
+* `xargs` v4.7 or later
+* `curl` v7 or later
+
+The Makefile is written with Debian-based systems in mind, including Ubuntu, and
+all those packages can be installed using `apt` from default repos. Other platforms
+ought to work fine, as long as these packages/versions can be installed. For example,
+MacOS is known to work, using `brew` to get a modern version of `bash` and GNU
+tools (there are many guides, [here is one](https://www.topbug.net/blog/2013/04/14/install-and-use-gnu-command-line-tools-in-mac-os-x/)).
+
+There are a couple make targets that call out to `apt` directly. You can avoid
+that requirement by pre-installing these packages using your own package manager,
+or simply do not invoke targets that rely on these packages:
+
+* `zip`
+* `nodejs`
+
+## Makefile "operating modes"
+
+When using the Makefile approach, the `include` directive is a shell command that
+retrieves the "bootstrap" Makefile from this project, saves it to the current working
+directory, and then echoes back the name of the saved file. Make reads the bootstrap
+Makefile and processes its directives and targets.
+
+The bootstrap Makefile is very barebones. Its primary purpose is to bootstrap this
+`tardigrade-ci` project so all the make targets are available. It does this through
+its own `include` directive. The bootstrap include directive processes a shell
+command with logic to determine whether to clone this project, and then echoes
+back the path to this project's Makefile.
+
+The bootstrap Makefile supports two "operating modes" that we're calling **user mode**
+and **developer mode**.
```

Comment:
Maybe reorder the explanation like below?

```suggestion
its own `include` directive.

The bootstrap Makefile supports two "operating modes" that we're calling **user mode** and **developer mode** 

The bootstrap include directive processes a shell command with logic to determine whether to operate in **user mode** or **developer mode**, and then echoes back the path to this project's Makefile.
```

Reply:
What if I just change `whether to clone this project` to `the operating mode (described below)`?

```suggestion
its own `include` directive. The bootstrap `include` directive processes a shell
command with logic to determine the operating mode (described below), and then echoes
back the path to the tardigrade-ci Makefile.

The bootstrap Makefile supports two "operating modes" that we're calling **user mode**
and **developer mode**.

```

Reply:
That works.